src/cmd/tools/m3ctl/client/http.go

```diff
@@ -0,0 +1,83 @@
+package client
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+const timeout = time.Duration(5 * time.Second)
+
+func DoGet(url string, getter func(reader io.Reader, zl *zap.Logger) error, zl *zap.Logger) error {
+	zl.Info(fmt.Sprintf("DoGet:url:%s:\n", url))
+	client := http.Client{
+		Timeout: timeout,
+	}
+	resp, err := client.Get(url)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		ioutil.ReadAll(resp.Body)
+		resp.Body.Close()
+	}()
+	if err := checkForAndHandleError(url, resp, zl); err != nil {
+		return err
+	}
+	return getter(resp.Body, zl)
+}
+
+func DoPost(url string, data io.Reader, getter func(reader io.Reader, zl *zap.Logger) error, zl *zap.Logger) error {
+	zl.Info(fmt.Sprintf("DoPost:url:%s:\n", url))
+	client := &http.Client{
+		Timeout: timeout,
+	}
+	req, err := http.NewRequest(http.MethodPost, url, data)
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		ioutil.ReadAll(resp.Body)
+		resp.Body.Close()
+	}()
+	if err := checkForAndHandleError(url, resp, zl); err != nil {
+		return err
+	}
+	return getter(resp.Body, zl)
+}
+
+func DoDelete(url string, getter func(reader io.Reader, zl *zap.Logger) error, zl *zap.Logger) error {
+	zl.Info(fmt.Sprintf("DoDelete:url:%s:\n", url))
```

Comment:
This should use fields, all log messages should start with lower capitalized letters as per Effective Go.
```
z.Info("request", zap.String("method", "delete"), zap.String("url", url))
```